t/006-escape.t

```diff
@@ -197,3 +197,100 @@ GET /lua
 %2C%24%40%7C%60
 --- no_error_log
 [error]
+
+
+
+=== TEST 15: escape type argument
+--- config
+    location /lua {
+        content_by_lua_block {
+            ngx.say(ngx.escape_uri("https://www.google.com", ngx.ESCAPE_URI))
+            ngx.say(ngx.escape_uri("https://www.google.com/query?q=test", ngx.ESCAPE_URI))
+            ngx.say(ngx.escape_uri("https://www.google.com/query?\r\nq=test", ngx.ESCAPE_URI))
+            ngx.say(ngx.escape_uri("-_.~!*'();:@&=+$,/?#", ngx.ESCAPE_URI))
+            ngx.say(ngx.escape_uri("<>[]{}\\\" ", ngx.ESCAPE_URI))
+        }
+    }
+--- request
+GET /lua
+--- response_body
+https://www.google.com
+https://www.google.com/query%3Fq=test
+https://www.google.com/query%3F%0D%0Aq=test
+-_.~!*'();:@&=+$,/%3F%23
+<>[]{}\"%20
+--- no_error_log
+[error]
+
+
+
+=== TEST 16: escape type argument
+--- config
+    location /lua {
+        content_by_lua_block {
+            ngx.say(ngx.escape_uri("https://www.google.com/?t=abc@ :", ngx.ESCAPE_URI))
+            ngx.say(ngx.escape_uri("https://www.google.com/?t=abc@ :", ngx.ESCAPE_ARGS))
+            ngx.say(ngx.escape_uri("https://www.google.com/?t=abc@ :", ngx.ESCAPE_URI_COMPONENT))
+            ngx.say(ngx.escape_uri("https://www.google.com/?t=abc@ :", ngx.ESCAPE_HTML))
+            ngx.say(ngx.escape_uri("https://www.google.com/?t=abc@ :", ngx.ESCAPE_REFRESH))
+            ngx.say(ngx.escape_uri("https://www.google.com/?t=abc@ :", ngx.ESCAPE_MEMCACHED))
+            ngx.say(ngx.escape_uri("https://www.google.com/?t=abc@ :", ngx.ESCAPE_MAIL_AUTH))
+        }
+    }
+--- request
+GET /lua
+--- response_body
+https://www.google.com/%3Ft=abc@%20:
+https://www.google.com/%3Ft=abc@%20:
+https%3A%2F%2Fwww.google.com%2F%3Ft%3Dabc%40%20%3A
+https://www.google.com/?t=abc@%20:
+https://www.google.com/?t=abc@%20:
+https://www.google.com/?t=abc@%20:
+https://www.google.com/?t=abc@%20:
+--- no_error_log
+[error]
+
+
+
+=== TEST 17: escape type out of range
+--- config
+    location /lua {
+        content_by_lua_block {
+            ngx.say(ngx.escape_uri("https://www.google.com", -1))
+        }
+    }
+--- request
+GET /lua
+--- error_code: 500
+--- error_log eval
+qr/\[error\] \d+#\d+: \*\d+ lua entry thread aborted: runtime error: "esc_type" \-1 out of range/
+
+
+
+=== TEST 18: escape type out of range
+--- config
+    location /lua {
+        content_by_lua_block {
+            ngx.say(ngx.escape_uri("https://www.google.com", 100))
+        }
+    }
+--- request
+GET /lua
+--- error_code: 500
+--- error_log eval
+qr/\[error\] \d+#\d+: \*\d+ lua entry thread aborted: runtime error: "esc_type" 100 out of range/
```

Comment:
The "esc_type" thing makes no sense to the end user. We should make it more user-friendly.